dss-collection-nft/contracts/DSSCollection.cdc

```diff
@@ -0,0 +1,626 @@
+        pub let id: UInt64
+        pub let collectionGroupID: UInt64
+        pub let logicalOperator: String // (AND / OR)
+        pub let typeName: String // (A.contractAddress.NFT...)
+        pub let slotType: String // (edition.id, edition.tier, play.id)
+        pub var items: {UInt64: UInt64}
+
+        // Add item to slot
+        //
+        access(contract) fun addItemToSlot(itemID: UInt64, itemValue: UInt64) {
+            pre {
+                DSSCollection.CollectionGroupData(
+                    id: self.collectionGroupID
+                ).open == true: "collection group not open"
+            }
+
+            self.items[itemID] = itemValue
+
+            emit ItemAddedToSlot(
+                 itemID: itemID,
+                 itemValue: itemValue,
+                 collectionGroupID: self.collectionGroupID
+             )
+        }
+
+        init (
+            collectionGroupID: UInt64,
+            logicalOperator: String,
+            typeName: String,
+            slotType: String
+        ) {
+            pre {
+                DSSCollection.CollectionGroupData(
+                    id: collectionGroupID
+                ).open == true: "collection group not open"
+            }
+
+            self.id = DSSCollection.nextSlotID
+            self.collectionGroupID = collectionGroupID
+            self.logicalOperator = logicalOperator
+            self.typeName = typeName
+            self.slotType = slotType
+            self.items = {}
+
+            DSSCollection.nextSlotID = self.id + 1 as UInt64
+
+            emit SlotCreated(
+                id: self.id,
+                collectionGroupID: self.collectionGroupID,
+                logicalOperator: self.logicalOperator,
+                typeName: self.typeName,
+                slotType: self.slotType
+            )
+        }
+    }
+
+    // A public struct to access CollectionGroup data
+    //
+    pub struct CollectionGroupData {
+        pub let id: UInt64
+        pub let name: String
+        pub let typeName: String
+        pub let open: Bool
+        pub let startTime: UFix64?
+        pub let endTime: UFix64?
+        pub let timeBound: Bool
+
+        init (id: UInt64) {
+            if let collectionGroup = &DSSCollection.collectionGroupByID[id] as &DSSCollection.CollectionGroup? {
+                self.id = collectionGroup.id
+                self.name = collectionGroup.name
+                self.typeName = collectionGroup.typeName
+                self.open = collectionGroup.open
+                self.startTime = collectionGroup.startTime
+                self.endTime = collectionGroup.endTime
+                self.timeBound = collectionGroup.timeBound
+            } else {
+                panic("collectionGroup does not exist")
+            }
+        }
+    }
+
+    // A top-level CollectionGroup with a unique ID and name
+    //
+    pub resource CollectionGroup {
+        pub let id: UInt64
+        pub let name: String
```

Comment:
do we want to include the longer description on chain too?

Reply:
Yes. I have added.